ruma-events-macros/src/collection_parse.rs

```diff
@@ -0,0 +1,62 @@
+//! Details of parsing input for the `event_content_collection` procedural macro.
+
+use syn::{
+    parse::{self, Parse, ParseStream},
+    Attribute, Expr, ExprLit, Ident, Lit, LitStr, Token,
+};
+
+mod kw {
+    syn::custom_keyword!(name);
+    syn::custom_keyword!(events);
+}
+
+/// The entire `event_content_collection!` macro structure directly as it appears in the source code..
+pub struct RumaCollectionInput {
+    /// Outer attributes on the field, such as a docstring.
+    pub attrs: Vec<Attribute>,
+
+    /// The name of the event.
+    pub name: Ident,
+
+    /// An array of valid matrix event types. This will generate the variants of the event content type "name".
+    /// There needs to be a corresponding variant in `ruma_events::EventType` for
+    /// this event (converted to a valid Rust-style type name by stripping `m.`, replacing the
+    /// remaining dots by underscores and then converting from snake_case to CamelCase).
+    pub events: Vec<LitStr>,
+}
+
+impl Parse for RumaCollectionInput {
+    fn parse(input: ParseStream<'_>) -> parse::Result<Self> {
+        let attrs = input.call(Attribute::parse_outer)?;
+        // name field
+        input.parse::<kw::name>()?;
+        input.parse::<Token![:]>()?;
+        // the name of our collection enum
+        let name: Ident = input.parse()?;
+        input.parse::<Token![,]>()?;
+
+        // events field
+        input.parse::<kw::events>()?;
+        input.parse::<Token![:]>()?;
+        // an array of event names `["m.room.whatever"]`
+        let mut events = vec![];
+        for item in input.parse::<syn::ExprArray>()?.elems.into_iter() {
+            if let Expr::Lit(ExprLit {
+                lit: Lit::Str(lit_str),
+                ..
+            }) = item
+            {
+                events.push(lit_str);
+            } else {
+                let msg = "values of field `events` are required to be a string literal by `event_content_collection!`";
```

Comment:
I think we don't need to tell the user of where this error is coming from if there's an error span that points right at it :)

```suggestion
                let msg = "values of field `events` are required to be a string literal";
```